### ISSUE TITLE
Error on save number non-unique fields

### ISSUE THREAD

**Issue by saostad** [vkarpov15/mongoose-unique-array#1]
**# when I want to update exist document it shows me this error:**

```
\projects\marketing-app\node_modules\mongoose-unique-array\index.js:103
      if (dirt.value._atomics && '$pushAll' in dirt.value._atomics) {
                    ^

TypeError: Cannot read property '_atomics' of null
    at $__dirty.forEach.dirt (C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\mongoose-unique-array\index.js:103:21)
    at Array.forEach (native)
    at model.<anonymous> (C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\mongoose-unique-array\index.js:102:21)
    at _next (C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\hooks-fixed\hooks.js:62:30)
    at fnWrapper (C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\hooks-fixed\hooks.js:186:8)
    at model.<anonymous> (C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\mongoose\lib\plugins\saveSubdocs.js:20:7)
    at _next (C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\hooks-fixed\hooks.js:62:30)
    at fnWrapper (C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\hooks-fixed\hooks.js:186:8)
    at C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\mongoose\lib\plugins\validateBeforeSave.js:33:13
    at C:\Users\Saeid\Desktop\projects\marketing-app\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

**# and this is my model:** 

```
//this file explain what the Project looks like!
var mongoose = require('mongoose');
var Schema = mongoose.Schema,
ObjectId = Schema.ObjectId;

const uniqueArrayPlugin = require('mongoose-unique-array');

var collectionName = "Project";

var projectModel = new Schema({
    title : {type: String},
    categories : [{ type : ObjectId, ref: 'Category', unique: true }],
    description : {type: String},
    owner : {type: String},
    location: {type: String},
    amount: {type: Number, default: 0},
    size: {type: Number, default: 0},
    startDate : {type: Date},
    endDate: {type: Date},
    files: [{
        address: String,
        description: String,
        type:String
        }],
    whatDid: {type: String},
    involvedCompanies: [{
        name: String,
        description: String
    }],
    involvedEmployees: [{
        employeeId : { type : ObjectId, ref: 'Employee' },
        role: String,
        description: String
    }],    
    capability:[{type: String, unique: true }],
    tags:[String]
});

// Add the unique array plugin
projectModel.plugin(uniqueArrayPlugin);

module.exports = mongoose.model(collectionName, projectModel, collectionName);
```

it happen when I want to update a numeric field like "amount" to null

**Comment by vkarpov15** [comment]
Below script executes without error for me. Please modify it to reproduce your issue:

```javascript
const arrayUnique = require('mongoose-unique-array');
const assert = require('assert');
const mongoose = require('mongoose');

const Schema = mongoose.Schema;
const ObjectId = mongoose.Schema.Types.ObjectId;

mongoose.connect('mongodb://localhost:27017/test');

var schema = new Schema({
        title : {type: String},
        categories : [{ type : ObjectId, ref: 'Category', unique: true }],
        description : {type: String},
        owner : {type: String},
        location: {type: String},
        amount: {type: Number, default: 0},
        size: {type: Number, default: 0},
        startDate : {type: Date},
        endDate: {type: Date},
        files: [{
            address: String,
            description: String,
            type:String
            }],
        whatDid: {type: String},
        involvedCompanies: [{
            name: String,
            description: String
        }],
        involvedEmployees: [{
            employeeId : { type : ObjectId, ref: 'Employee' },
            role: String,
            description: String
        }],
        capability:[{type: String, unique: true }],
        tags:[String]
    });

    schema.plugin(arrayUnique);
    var M = mongoose.model('T6', schema);
    var m = new M({
      number: 1,
      categories: null
    });

    m.save(function(error) {
      assert.ifError(error);
      M.update({}, { $set: { number: 0 } }, function(error) {
        assert.ifError(error);
        console.log('done');
      });
    });
```

**Comment by g-div** [comment]
I'm experiencing the same issue, this happens when for example I try to delete a relation by setting the relation field to `null` (or `undefined`). Here is the modified repro-script:

```
const arrayUnique = require('mongoose-unique-array');
const assert = require('assert');
const mongoose = require('mongoose');

const Schema = mongoose.Schema;
const ObjectId = mongoose.Schema.Types.ObjectId;

mongoose.connect('mongodb://localhost:27017/test');

var schema = new Schema({
        title : {type: String},
        categories : [{ type : ObjectId, ref: 'Category', unique: true }],
        description : {type: String},
        owner : {type: String},
        location: {type: String},
        amount: {type: Number, default: 0},
        size: {type: Number, default: 0},
        startDate : {type: Date},
        endDate: {type: Date},
        files: [{
            address: String,
            description: String,
            type:String
            }],
        whatDid: {type: String},
        involvedCompanies: [{
            name: String,
            description: String
        }],
        involvedEmployees: [{
            employeeId : { type : ObjectId, ref: 'Employee' },
            role: String,
            description: String
        }],
        capability:[{type: String, unique: true }],
        tags:[String]
    });

    schema.plugin(arrayUnique);
    var M = mongoose.model('T6', schema);
    var m = new M({
      number: 1,
      categories: '5c62b96b3f3f340f428c304d'
    });

    m.save(function(error) {
      assert.ifError(error);

      // remove the categories relation will cause the error
      m.categories = null;
      m.save(function(error) {
        assert.ifError(error);
        console.log('done');
      });
    });
```

**Comment by vkarpov15** [comment]
@g-div thanks! Fix is released in 0.3.1 :+1:
